ui/src/app/base/components/DhcpForm/DhcpForm.tsx

```diff
@@ -89,9 +93,7 @@ export const DhcpForm = ({
   }
 
   return (
-    <FormikForm<DHCPFormValues>
-      cleanup={dhcpsnippetActions.cleanup}
-      errors={errors}
+    <Formik
```

Comment:
Do you think not including the form values generic here could cause an issue? `Formik` still infers the values from `initialValues` but I'm wondering if there are any instances where that's not enough. Can you think of a way where we could include the generic once, and have it be used for both `Formik` and `FormikFormContent`?

Reply:
I did end up having to set the value type for both components in a few places as it wasn't able to infer the full type (e.g. [here](https://github.com/canonical-web-and-design/maas-ui/pull/3812/files#diff-447826ada27bb574ed7ccc0695504cd55577b1e7ac1a7c8a500d3c0480a9dc6fR98)).

I don't think there's any way of providing the type automatically, I guess we could have a pattern where if you want to share the values type then you use an alias e.g. `type FormValues = AddDeviceValues`.

I suppose the danger with this new way is that errors could creep in if you accidentally used two different generics for the two components.

I guess the way to solve that would be two wrap both components in a new component that passes the generic along, maybe we could call it something like `FormikForm` :).